### ISSUE TITLE
circuits equal up to global phase

### ISSUE THREAD

**Issue by punkdit** [zxcalc/pyzx#143]
```
c0 = zx.Circuit(1)
c0.add_gate("NOT", 0)
c0.add_gate("Z", 0)

c1 = zx.Circuit(1)
c1.add_gate("Z", 0)
c1.add_gate("NOT", 0)
print(c0.verify_equality(c1))
```

It should be made clear in the documentation that `verify_equality` is equality of circuits up to global phase only. Or have I done something wrong?

See also: https://quantumcomputing.stackexchange.com/a/25789

**Comment by jvdwetering** [comment]
Yes, that's right, this only verifies equality up to global phase.
It wouldn't be too hard to add support for testing exact equality here. Would that be useful for your use-case?

**Comment by punkdit** [comment]
Thankyou John. The use case is to generate and/or compute efficiently products and equality within the Clifford group. (The tablaux methods can only do this up to global phase.) As mentioned in the stackexchange question, one idea is to compute with normal forms of ZX diagrams that represent these Clifford elements. How would I do this with PyZX ?

**Comment by jvdwetering** [comment]
Do you want to work with states or with unitaries? And do the normal forms have to be unique? It is possible to get a unique normal form for states using ZX, but this is currently not implemented in PyZX. To do that you would use the algorithm described in https://arxiv.org/abs/2306.05204 but adapted again to qubits.

**Comment by punkdit** [comment]
I'm interested in the Clifford group: these are unitary operators, but they should have efficient (in qubit number) representations. It sounds like you are telling me this is an open problem. And there needs to be a unique normal form for equality testing.

**Comment by akissinger** [comment]
There seems to be some miscommunication here. It is trivial to add an optional parameter to check the global phase is zero. This is done now. One I commit, you can call `c0.verify_equality(c1, up_to_global_phase=False)`, which will return false.
